nym-vpn-core/nym-vpn-lib/src/lib.rs

```diff
@@ -818,6 +820,9 @@ async fn run_nym_vpn(
         Err(err) => {
             error!("Nym VPN returned error: {err}");
             debug!("{err:?}");
+            set_listener_status(StatusEvent::Exit(ExitStatus::Failed {
```

Comment:
Do we really want to be setting these uniffi specific status events unconditionally? If I understand it correctly it's setting a status event that is only used by uniffi clients, but it's kinda handled implicitly for other clients by the running instance not existing?

I'm thinking can we rename this to something like `uniffi_set_listener_status` to emphasize that this is ignored as far as non-uniffi clients are concerned? 

Related, probably just me being silly, but why do we need to even set this? Should the uniffi stuff listen on vpn_exit_rx for this?